app/forms/catalog_record_id_form.rb

```diff
@@ -0,0 +1,100 @@
+  def persisted?
+    false
+  end
+
+  def setup_properties!(_options)
+    object_catalog_record_ids = model.identification.catalogLinks.filter_map { |catalog_link| catalog_link if catalog_link.catalog == CatalogRecordId.value }
+
+    self.catalog_record_ids = object_catalog_record_ids.map { |catalog_record_id| CatalogRecordIdForm::Row.new(value: catalog_record_id.catalogRecordId, refresh: catalog_record_id.refresh) }
+
+    other_object_catalog_record_ids = model.identification.catalogLinks.filter_map { |catalog_link| catalog_link if catalog_link.catalog != CatalogRecordId.value }
+
+    self.other_catalog_record_ids = other_object_catalog_record_ids.map { |catalog_record_id| CatalogRecordIdForm::Row.new(value: catalog_record_id.catalogRecordId, refresh: catalog_record_id.refresh, catalog: catalog_record_id.catalog) }
+  end
+
+  def unique_catalog_record_id_value
+    # each catalog record ID must be unique
+    catalog_record_id_values = catalog_record_ids.map(&:value)
+    errors.add(:catalog_record_id, "must be unique") if catalog_record_id_values.size != catalog_record_id_values.uniq.size
+  end
+
+  def valid_catalog_record_id_value
+    # must match the expected pattern
+    errors.add(:catalog_record_id, "must be in an allowed format") unless CatalogRecordId.valid?(catalog_record_ids.map(&:value))
+  end
+
+  def single_catalog_record_id_refresh
+    # at most one catalog record ID per catalog (Symphony/Folio) can be set to refresh == true
+    errors.add(:refresh, "is only allowed for a single catalog record ID.") if catalog_record_ids.count { |id| id.refresh && id._destroy != "1" } > 1
+  end
+
+  # this is overriding Reforms save method, since we are persisting catalog record IDs in cocina only
+  def save_model
+    # fetch the existing previous catalog record ID values from the cocina object
+    existing_previous_catalog_record_ids = CatalogRecordId.new(model).previous_links
+
+    # these are all of the existing catalog record ID values the user wants to remove (i.e. for which they clicked the trash icon)
+    removed_catalog_record_ids = catalog_record_ids.filter_map { |id| id.value if id._destroy == "1" }
+
+    # build an array of all previous catalog record IDs (which includes the existing previous catalog record IDs plus any newly removed ones)
+    updated_previous_catalog_record_ids = (existing_previous_catalog_record_ids + removed_catalog_record_ids).map do |id_value|
+      {catalog: CatalogRecordId.previous_value, catalogRecordId: id_value, refresh: false}
+    end.uniq
+
+    updated_catalog_record_ids = catalog_record_ids.filter_map do |id|
+      {catalog: CatalogRecordId.value, catalogRecordId: id.value, refresh: id.refresh} unless id._destroy == "1"
+    end
+
+    other_existing_catalog_record_ids = other_catalog_record_ids.map do |id|
+      {catalog: id.catalog, catalogRecordId: id.value, refresh: id.refresh}
+    end
```

Comment:
Added this to help persist Folio catalog links when in Symphony mode, and vice versa.